### PR TITLE
Gets rid of UNIT_TESTS compiler warning

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Basically every single CI Run is throwing the following warning:
```txt
code/modules/unit_tests/lootpanel.dm:24:warning (unused_var): new_box: variable defined but not used
```

You may find an example here: https://github.com/tgstation/tgstation/actions/runs/8698627681/job/23855921813#step:9:24

This is pretty silly but I don't really know why we even have this variable (I assume there's something far more complicated underneath the surface based on what the unit test is asserting), but assuming it is important let's just insert a do-nothing procedure to get rid of the compiler warning while ensuring the unit test is actually operating as it should.

I also don't really like the fact that this is a warning instead of an error but let's tackle this problem one step at a time by at least getting rid of the compiler warning in a quick advance PR while I dwell on this issue (is there a way to get the Dreamchecker linter to look at the unit test files? it's caught perfectly fine in the langserver)

### CODE REVIEW

File: code/modules/unit_tests/lootpanel.dm [hunk 1]
```diff
@@ -32,3 +32,5 @@
 	TEST_ASSERT_EQUAL(length(panel.contents), 2, "Panel shouldnt dupe searchables if reopened")
 
 	mock_client.mob = null
+	if(!isnull(new_box)) // this is effectively a do-nothing just so we can get the compiler to shut up, `allocate()` will ensure it gets qdel'd
+		new_box = null
```
Comment (MrMelbert) [inline code/modules/unit_tests/lootpanel.dm:36]:
```suggestion
	pass(new_box)
```
If it's a do-nothing, then we can just make it a do-nothing

**Comment by san7890** [comment]
i think the real fix is the whole `Unimplemented proc & var warnings are currently suppressed` OpenDream footgun since that surely would have caught it since we compile OpenDream with CI flags enabled... but that's definitely a really hard project isn't it

**Comment by EvilDragonfiend** [comment]
```dm
	var/obj/item/storage/toolbox/new_box
	new_box = new_box = allocate(/obj/item/storage/toolbox, one_over)
```
This looks silly, but it works too.

**Comment by san7890** [comment]
i think the `pass()` is my favorite since it's built for that purpose but thanks for sharing

**Comment by Mothblocks** [comment]
I don't understand. If we're not using the variable, why are we naming it at all?

**Comment by san7890** [comment]
i didn't know what the RPG loot subsystem was when i published this PR, it's just checking to make sure things on adjacent tiles don't add to the readout for the RPG loot window

**Comment by jlsnow301** [comment]
found the problem (it's brain damage)
```
allocate(/obj/item/storage/toolbox, one_over)
```
just remove var pls

**Pull request merged**